sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -2975,22 +2975,63 @@ def hpat_pandas_series_idxmax(self, axis=None, skipna=True):
     if not isinstance(self.data.dtype, types.Number):
         ty_checker.raise_exc(self.data.dtype, 'int, float', 'self.data.dtype')
 
-    if not (isinstance(skipna, (types.Omitted, types.Boolean, bool)) or skipna is True):
+    if not (isinstance(skipna, (types.Omitted, types.Boolean, bool)) or skipna is None):
         ty_checker.raise_exc(skipna, 'bool', 'skipna')
 
     if not (isinstance(axis, types.Omitted) or axis is None):
         ty_checker.raise_exc(axis, 'None', 'axis')
 
     if isinstance(self.index, types.NoneType) or self.index is None:
-        def hpat_pandas_series_idxmax_impl(self, axis=None, skipna=True):
-            return numpy.argmax(self._data)
+        if isinstance(self.data, StringArrayType):
+            def hpat_pandas_series_idxmax_impl(self, axis=None, skipna=None):
+                if skipna is None:
+                    _skipna = True
+                else:
+                    raise ValueError("Method idxmax(). Unsupported parameter 'skipna'=False with str data")
+
+                return numpy.argmax(self._data)
+
+            return hpat_pandas_series_idxmax_impl
+
+        def hpat_pandas_series_idxmax_impl(self, axis=None, skipna=None):
+            # return numpy.argmax(self._data)
+            if skipna is None:
+                _skipna = True
+            else:
+                _skipna = skipna
+
+            if _skipna:
+                return numpy_like.nanargmax(self._data)
+
+            return numpy_like.argmax(self._data)
```

Comment:
I would create variable `none_index = isinstance(self.index, types.NoneType) or self.index is None`, then make common implementations `hpat_pandas_series_idxmax_impl` and `hpat_pandas_series_idxmax_impl` for both cases with index and without index. In the implementations I would add something like that:
```python
if none_index == True:  # noqa
    return result
else:
    self._index[int(result)]
```

The same is applicable for `idxmin`.